bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WLedHandler.java

```diff
@@ -0,0 +1,439 @@
+                            && hasWhite) {
+                        // Google sends this when it wants white
+                        sendWhite();
+                    } else {
+                        sendGetRequest(
+                                "/win&TT=1000&FX=0&CY=0&CL=" + createColorHex(primaryColor) + "&A=" + masterBrightness);
+                    }
+                } else {// should only be PercentType left
+                    masterBrightness = new BigDecimal(command.toString()).multiply(new BigDecimal(2.55));
+                    sendGetRequest("/win&TT=1000&A=" + masterBrightness);
+                }
+                return;
+            case CHANNEL_PRIMARY_COLOR:
+                if (command instanceof OnOffType) {
+                    return;
+                } else if (command instanceof HSBType) {
+                    primaryColor = new HSBType(command.toString());
+                    sendGetRequest("/win&CL=" + createColorHex(primaryColor));
+                } else if (command instanceof IncreaseDecreaseType) {
+                    return;
+                } else {// Percentype
+                    primaryColor = new HSBType(primaryColor.getHue().toString() + ","
+                            + primaryColor.getSaturation().toString() + ",command");
+                    sendGetRequest("/win&CL=" + createColorHex(primaryColor));
+                }
+                return;
+            case CHANNEL_SECONDARY_COLOR:
+                if (command instanceof OnOffType) {
+                    return;
+                } else if (command instanceof HSBType) {
+                    secondaryColor = new HSBType(command.toString());
+                    sendGetRequest("/win&C2=" + createColorHex(secondaryColor));
+                } else if (command instanceof IncreaseDecreaseType) {
+                    return;
+                } else {// Percentype
+                    secondaryColor = new HSBType(secondaryColor.getHue().toString() + ","
+                            + secondaryColor.getSaturation().toString() + ",command");
+                    sendGetRequest("/win&C2=" + createColorHex(secondaryColor));
+                }
+                return;
+            case CHANNEL_PALETTES:
+                sendGetRequest("/win&FP=" + command);
+                break;
+            case CHANNEL_FX:
+                sendGetRequest("/win&FX=" + command);
+                break;
+            case CHANNEL_SPEED:
+                BigDecimal bigTemp = new BigDecimal(command.toString());
+                if (OnOffType.OFF.equals(command)) {
+                    bigTemp = BigDecimal.ZERO;
+                } else if (OnOffType.ON.equals(command)) {
+                    bigTemp = new BigDecimal(255);
+                } else {
+                    bigTemp = new BigDecimal(command.toString()).multiply(new BigDecimal(2.55));
+                }
```

Comment:
What you are doing here is kinda inefficient with all of the new object creation going on.
If you take advantage of the built-in type conversion functions in State you can make this much more concise.
```suggestion
                BigDecimal bigTemp = ((State)command).as(PercentType.class).multiply(BIG_DECIMAL_2_55));
```

The same change would apply for the CHANNEL_INTENSITY case as well.